libs/sr/solver/src/solver.ts

```diff
@@ -121,48 +125,119 @@ export class Solver {
     this.worker.terminate()
   }
 
-  private detachNodes() {
+  private detachAndPruneNodes() {
     // Step 2: Detach nodes from Calculator
     const relicSetKeys = new Set(allRelicSetKeys)
     const lightConeKeys = new Set(allLightConeKeys)
-    const detachedNodes = detach(
-      [
-        // team
-        sum(
-          ...this.frames.map((frame, i) =>
-            prod(
-              frame.multiplier,
-              new Read(frame.tag, 'sum').with('preset', `preset${i}` as Preset)
-            )
+    const nodes = [
+      // stat filters
+      ...this.statFilters.map(({ tag }) => new Read(tag, 'sum')),
```

Comment:
You also need to flip the sign on the read node for max constraints, e.g., with `prod(-1, <read node>)`